src/io/ipc/read/reader.rs

```diff
@@ -36,6 +36,16 @@ pub struct FileMetadata {
     pub(crate) size: u64,
 }
 
+/// prepare `scratch` to read the message
+pub(super) fn prepare_scratch(scratch: &mut Vec<u8>, message_length: usize) -> &mut [u8] {
+    // ensure that we have enough scratch space
+    if message_length > scratch.len() {
+        scratch.resize(message_length, 0);
+    }
```

Comment:
I wonder if

```suggestion
    if message_length > scratch.len() {
        *scratch = vec![0; message_length];
    }
```

wouldn't be even faster. i.e. instead of having to re-alloc old data that will be overwritten anyways, we call `alloc_zeroed` (the min requirements for safety in `Read`).

We do something similar in parquet2: https://github.com/jorgecarleitao/parquet2/blob/main/src/read/page/reader.rs#L199

Reply:
I think you are right. I suggest to first check the capacity. If the new size fits in capacity we only append the `new_size - size` zeros.

And otherwise go for the `alloc_zeroed` strategy.

Reply:
Or maybe one step further may be to:

```rust
if length > scratch.capacity() {
    *scratch = vec![0; message];
} else length > scratch.len() {
    scratch.resize(length, 0)
} else {
    scratch.truncate(length)
}
scratch
```

so, if we need to alloc, we just alloc-zero. If we only need a bit more zeroed space, we zeroed it.

Reply:
> Or maybe one step further may be to:
> 
> ```rust
> if length > scratch.capacity() {
>     *scratch = vec![0; message];
> } else if length > scratch.len() {
>     scratch.resize(length, 0);
> } else {
>     scratch.truncate(length);
> }
> scratch
> ```
> 
> so, if we need to alloc, we just alloc-zero. If we only need a bit more zeroed space, we zeroed it.

Haha.. my thougts in code! ;)

Reply:
we would do a bunch of resize and truncates, since every message will be slightly different length. In this context we could skip the `truncate` and only expose `&mut scratch[..length]` as this PR already does.

We just need to be mindful that data on the scratch can be read between messages if the message is not valid (a bit like out of bound reads but for owned). imo this is not an issue because the user already offered `scratch` to the reader and should thus assume that `Read` may read any of that content.